mat64/dense_arithmetic.go

```diff
@@ -124,12 +124,21 @@ func (m *Dense) Norm(ord float64) float64 {
 func (m *Dense) Add(a, b Matrix) {
 	ar, ac := a.Dims()
 	br, bc := b.Dims()
-
 	if ar != br || ac != bc {
 		panic(ErrShape)
 	}
 
+	aMat, _ := untranspose(a)
```

Comment:
Yea, just trying to keep as much as possible from the original Add function so this commit is mostly about list_test.